Source/WebCore/platform/graphics/skia/ImageBufferSkiaAcceleratedBackend.cpp

```diff
@@ -135,6 +147,26 @@ void ImageBufferSkiaAcceleratedBackend::waitForAcceleratedRenderingFenceCompleti
     m_fence = nullptr;
 }
 
+RefPtr<ImageBuffer> ImageBufferSkiaAcceleratedBackend::copyAcceleratedImageBufferBorrowingBackendRenderTarget(const ImageBuffer* imageBuffer) const
+{
+    auto* glContext = PlatformDisplay::sharedDisplay().skiaGLContext();
+    if (!glContext || !glContext->makeContextCurrent())
+        return nullptr;
+
+    auto* grContext = PlatformDisplay::sharedDisplay().skiaGrContext();
+    RELEASE_ASSERT(grContext);
+
+    auto backendRenderTarget = SkSurfaces::GetBackendRenderTarget(m_surface.get(), SkSurfaces::BackendHandleAccess::kFlushRead);
+
+    const auto& imageInfo = m_surface->imageInfo();
+    auto surface = SkSurfaces::WrapBackendRenderTarget(grContext, backendRenderTarget, kTopLeft_GrSurfaceOrigin, imageInfo.colorType(), imageInfo.refColorSpace(), &m_surface->props());
+    if (!surface || !surface->getCanvas())
+        return nullptr;
+
+    auto backend = ImageBufferSkiaAcceleratedBackend::create(parameters(), { }, WTFMove(surface));
```

Comment:
So, this new create() is only used here, I guess we could just call new here, no?

Reply:
The original create() function uses the new constructor as well:

```
std::unique_ptr<ImageBufferSkiaAcceleratedBackend> ImageBufferSkiaAcceleratedBackend::create(const Parameters& parameters, const ImageBufferCreationContext& creationContext)
{
....
    auto surface = SkSurfaces::RenderTarget(grContext, skgpu::Budgeted::kNo, imageInfo, PlatformDisplay::sharedDisplay().msaaSampleCount(), kTopLeft_GrSurfaceOrigin, &properties);
....
    return create(parameters, creationContext, WTFMove(surface));
}
```

Therefore I think it's better to keep the `create()` variant taking a `SkSurface&&`.